plugin/src/main/cpp/extensions/openxr_fb_color_space_extension_wrapper.cpp

```diff
@@ -0,0 +1,291 @@
+	BIND_ENUM_CONSTANT(COLOR_SPACE_ADOBE_RGB)
+	BIND_ENUM_CONSTANT(COLOR_SPACE_RUNTIME_DEFAULT)
+}
+
+void OpenXRFbColorSpaceExtensionWrapper::cleanup() {
+	fb_color_space_ext = false;
+	supported_color_spaces_set = false;
+}
+
+void OpenXRFbColorSpaceExtensionWrapper::populate_supported_color_spaces() {
+	if (supported_color_spaces_set) {
+		return;
+	}
+
+	uint32_t num_color_spaces = 0;
+	XrResult result = xrEnumerateColorSpacesFB(SESSION, 0, &num_color_spaces, nullptr);
+	if (XR_FAILED(result)) {
+		UtilityFunctions::printerr("OpenXR: Failed to get supported color space count [", get_openxr_api()->get_error_string(result), "]");
+		return;
+	}
+
+	supported_color_spaces.resize(num_color_spaces);
+
+	result = xrEnumerateColorSpacesFB(SESSION, num_color_spaces, &num_color_spaces, supported_color_spaces.ptrw());
+	if (XR_FAILED(result)) {
+		UtilityFunctions::printerr("OpenXR: Failed to get supported color spaces [", get_openxr_api()->get_error_string(result), "]");
+		supported_color_spaces.clear();
+		return;
+	}
+
+	supported_color_spaces_set = true;
+}
+
+OpenXRFbColorSpaceExtensionWrapper::ColorSpace OpenXRFbColorSpaceExtensionWrapper::from_openxr_color_space(XrColorSpaceFB p_color_space) {
+	switch (p_color_space) {
+		case XR_COLOR_SPACE_UNMANAGED_FB: {
+			return COLOR_SPACE_UNMANAGED;
+		} break;
+		case XR_COLOR_SPACE_REC2020_FB: {
+			return COLOR_SPACE_REC2020;
+		} break;
+		case XR_COLOR_SPACE_REC709_FB: {
+			return COLOR_SPACE_REC709;
+		} break;
+		case XR_COLOR_SPACE_RIFT_CV1_FB: {
+			return COLOR_SPACE_RIFT_CV1;
+		} break;
+		case XR_COLOR_SPACE_RIFT_S_FB: {
+			return COLOR_SPACE_RIFT_S;
+		} break;
+		case XR_COLOR_SPACE_QUEST_FB: {
+			return COLOR_SPACE_QUEST;
+		} break;
+		case XR_COLOR_SPACE_P3_FB: {
+			return COLOR_SPACE_P3;
+		} break;
+		case XR_COLOR_SPACE_ADOBE_RGB_FB: {
+			return COLOR_SPACE_ADOBE_RGB;
+		} break;
+		case XR_COLOR_SPACE_MAX_ENUM_FB:
+		default: {
+			ERR_FAIL_V_MSG(COLOR_SPACE_RUNTIME_DEFAULT, vformat("Unknown OpenXR color space: %s", p_color_space));
+		}
+	}
+}
+
+XrColorSpaceFB OpenXRFbColorSpaceExtensionWrapper::to_openxr_color_space(ColorSpace p_color_space) {
+	switch (p_color_space) {
+		case COLOR_SPACE_UNMANAGED: {
+			return XR_COLOR_SPACE_UNMANAGED_FB;
+		} break;
+		case COLOR_SPACE_REC2020: {
+			return XR_COLOR_SPACE_REC2020_FB;
+		} break;
+		case COLOR_SPACE_REC709: {
+			return XR_COLOR_SPACE_REC709_FB;
+		} break;
+		case COLOR_SPACE_RIFT_CV1: {
+			return XR_COLOR_SPACE_RIFT_CV1_FB;
+		} break;
+		case COLOR_SPACE_RIFT_S: {
+			return XR_COLOR_SPACE_RIFT_S_FB;
+		} break;
+		case COLOR_SPACE_QUEST: {
+			return XR_COLOR_SPACE_QUEST_FB;
+		} break;
+		case COLOR_SPACE_P3: {
+			return XR_COLOR_SPACE_P3_FB;
+		} break;
+		case COLOR_SPACE_ADOBE_RGB: {
+			return XR_COLOR_SPACE_ADOBE_RGB_FB;
+		} break;
+		case COLOR_SPACE_RUNTIME_DEFAULT:
+		default: {
+			ERR_FAIL_V_MSG(XR_COLOR_SPACE_UNMANAGED_FB, vformat("Unknown color space: %s", p_color_space));
```

Comment:
We shouldn't default to `XR_COLOR_SPACE_UNMANAGED_FB` since this applies different settings than the runtime default.
Can we just return `XR_COLOR_SPACE_MAX_ENUM_FB` instead?